core/engine/src/module/source.rs

```diff
@@ -241,14 +241,77 @@ impl std::fmt::Debug for SourceTextModule {
 #[derive(Debug)]
 struct ModuleCode {
     has_tla: bool,
-    requested_modules: IndexSet<JsString, BuildHasherDefault<FxHasher>>,
+    requested_modules: IndexSet<super::ModuleRequest, BuildHasherDefault<FxHasher>>,
     source: boa_ast::Module,
     source_text: SourceText,
     path: Option<PathBuf>,
     import_entries: Vec<ImportEntry>,
     local_export_entries: Vec<LocalExportEntry>,
     indirect_export_entries: Vec<IndirectExportEntry>,
-    star_export_entries: Vec<JsString>,
+    star_export_entries: Vec<super::ModuleRequest>,
+}
+
+struct ModuleRequestsVisitor<'a> {
+    interner: &'a Interner,
+    requests: IndexSet<super::ModuleRequest, BuildHasherDefault<FxHasher>>,
+}
+
+impl<'ast> boa_ast::visitor::Visitor<'ast> for ModuleRequestsVisitor<'_> {
+    type BreakTy = std::convert::Infallible;
+
+    fn visit_import_declaration(
+        &mut self,
+        node: &'ast boa_ast::declaration::ImportDeclaration,
+    ) -> std::ops::ControlFlow<Self::BreakTy> {
+        let specifier = node.specifier().sym().to_js_string(self.interner);
+        let attributes = node
+            .attributes()
+            .iter()
```

Comment:
thought: this `iter().map(|attr|(JsString, JsString)).collect()` happens multiple times in the code base and may benefit from a new type

I think I've seen it at least 3 or 4 times, so it's definitely a candidate. But I'll leave that up to you all